src/khoj/main.py

```diff
@@ -118,6 +112,15 @@ def run():
 
         gui.exec()
 
+    if not state.demo:
+        # Setup task scheduler
+        poll_task_scheduler()
+
+    # Start Server
+    configure_server(args, required=False)
+    configure_routes(app)
+    start_server(app, host=args.host, port=args.port, socket=args.socket)
+
```

Comment:
This block should be put under `if not args.gui` to prevent it being called twice in `gui` mode. I suggest we keep the ordering of code execution same as before. The app start sequence is sensitive to change because of the multiple threads (GUI, polling thread)

```suggestion
    # Start Server
    configure_server(args, required=False)
    configure_routes(app)
    start_server(app, host=args.host, port=args.port, socket=args.socket)

```